File: test/fixtures/pysample/src/payment/validator.py
from .engine import compute_fee

SUPPORTED = ['USD', 'EUR', 'INR']
RISK_THRESHOLD = 80


def validate_transaction(amount, currency, method):
    if amount <= 0:
        raise ValueError('amount must be positive')
    if currency not in SUPPORTED:
        raise ValueError('unsupported currency')
    if amount > method['limit']:
        raise ValueError('exceeds method limit')
    fee = compute_fee(amount)
    risk = assess_risk(amount, method)
    if risk > RISK_THRESHOLD:
        return {'ok': False, 'reason': 'risk', 'fee': fee}
    return {'ok': True, 'reason': None, 'fee': fee}


def assess_risk(amount, method):
    base = 60 if amount > 10000 else 20
    if method['kind'] == 'card':
        base += 10
    return base

File: test/fixtures/sample/src/payment/validator.ts
import { computeFee } from './engine.js';

interface PaymentMethod {
  limit: number;
  kind: string;
}

interface ValidationResult {
  ok: boolean;
  reason: string | null;
  fee: number;
}

const SUPPORTED_CURRENCIES = ['USD', 'EUR', 'INR'];
const RISK_THRESHOLD = 80;

export function validateTransaction(amount: number, currency: string, method: PaymentMethod): ValidationResult {
  if (amount <= 0) {
    throw new Error('amount must be positive');
  }
  if (!SUPPORTED_CURRENCIES.includes(currency)) {
    throw new Error('unsupported currency');
  }
  if (amount > method.limit) {
    throw new Error('exceeds method limit');
  }
  const fee = computeFee(amount);
  const riskScore = assessRisk(amount, method);
  if (riskScore > RISK_THRESHOLD) {
    return { ok: false, reason: 'risk', fee: fee };
  }
  return { ok: true, reason: null, fee: fee };
}

function assessRisk(amount: number, method: PaymentMethod): number {
  let base = amount > 10000 ? 60 : 20;
  if (method.kind === 'card') {
    base += 10;
  }
  return base;
}
